src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework.UnitTests/ExampleTests.cs

```diff
@@ -18,6 +18,26 @@ namespace Microsoft.CommonLanguageServerProtocol.Framework.UnitTests;
 
 public partial class ExampleTests
 {
+    [Fact]
+    public async Task InitializeServer_WithMultipleHandlerRegistration_Succeeds()
+    {
+        // Arrange
+        var logger = GetLogger();
+        var server = TestExampleLanguageServer.CreateLanguageServer(logger);
+
+        // Act
+        // Verifying that this does not throw.
+        var _ = await server.InitializeServerAsync();
+
+        // Assert
+        var handlerProvider = server.GetTestAccessor().GetQueueAccessor()!.Value.GetHandlerProvider();
+        var methods = handlerProvider.GetRegisteredMethods();
```

Comment:
>var methods = handlerProvider.GetRegisteredMethods(); [](http://example.com/codeflow?start=8&length=53)

might be nice to assert that there are only three methods registered too

Reply:
It's actually more than 3 because of the handlers already registered. I'll probably just do a full `Assert.Collection` but I was on my way out the door as I was finishing that and just wanted to verify it worked in the base case so my brain could let it go.